cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/MapTaggedOutputWritable.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.etl.batch.mapreduce;
+
+import co.cask.cdap.etl.batch.conversion.WritableConversions;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+import org.apache.hadoop.io.WritableComparable;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+/**
+ * Map output for which includes stageName and the record provided. This can be used to tag map output with stageName.
+ * @param <RECORD>
+ */
+public class MapTaggedOutputWritable<RECORD> implements WritableComparable<MapTaggedOutputWritable> {
+  private String stageName;
+  private RECORD record;
+
+  // required by Hadoop
+  @SuppressWarnings("unused")
+  public MapTaggedOutputWritable() {
+  }
+
+  public MapTaggedOutputWritable(String stageName, RECORD record) {
+    this.stageName = stageName;
+    this.record = record;
+  }
+
+  public String getStageName() {
+    return stageName;
+  }
+
+  public RECORD getRecord() {
+    return record;
+  }
+
+  public void setStageName(String stageName) {
+    this.stageName = stageName;
+  }
+
+  public void setRecord(RECORD record) {
+    this.record = record;
+  }
+
+  // TODO change this - use stageNAme
```

Comment:
shouldn't be a todo. if equals is using stage name, this needs to compare with stage name too.